pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -368,6 +368,16 @@ func (service *AlertRuleService) ReplaceRuleGroup(ctx context.Context, user iden
 		return err
 	}
 
+	for _, rule := range group.Rules {
```

Comment:
NIT: Validating early has some performance benefits, but I wonder if we shouldn't just call `ValidateUID` during `DBstore.InsertAlertRules(...)` in the alternative to generating the UID. Central and guaranteed to prevent inserting invalid UIDs regardless of the path taken to insertion.